scripts/datascience/themes_quality/wiki_puzzle_gemini_pipeline/mp_puzzle_generation/RateLimiter.py

```
@@ -0,0 +1,183 @@
+            'requests_made': 0,
+            'requests_blocked_rpm': 0,
+            'requests_blocked_interval': 0,
+            'requests_blocked_concurrent': 0,
+            'total_wait_time': 0.0,
+            'longest_wait': 0.0,
+            'start_time': time.time()
+        }
+        self.total_requests = 0
+
+
+def create_rate_limiter(multiprocessing_enabled: bool = True, **kwargs) -> 'SharedRateLimiter | LocalRateLimiter':
```

Ruff (0.12.2)

Undefined name `SharedRateLimiter`

(F821)